cosmwasm/contracts/crosschain-registry/src/contract.rs

```diff
@@ -0,0 +1,239 @@
+#[cfg(not(feature = "imported"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{to_binary, Binary, Deps, DepsMut, Env, MessageInfo, Response, StdResult};
+use cw2::set_contract_version;
+
+use crate::error::ContractError;
+use crate::msg::{ExecuteMsg, InstantiateMsg, QueryMsg};
+use crate::state::{
+    State, CHAIN_TO_CHAIN_CHANNEL_MAP, CHANNEL_ON_CHAIN_CHAIN_MAP, CONTRACT_ALIAS_MAP, STATE,
+};
+use crate::{execute, Registries};
+
+// version info for migration
+const CONTRACT_NAME: &str = "crates.io:crosschain-registry";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "imported"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    // validate owner address and save to state
+    let owner = deps.api.addr_validate(&msg.owner)?;
+    let state = State { owner };
+    STATE.save(deps.storage, &state)?;
+
+    Ok(Response::new().add_attribute("method", "instantiate"))
+}
+
+#[cfg_attr(not(feature = "imported"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        // Contract aliases
+        ExecuteMsg::ModifyContractAlias { operations } => {
+            execute::contract_alias_operations(deps, operations)
+        }
+
+        // Chain channel links
+        ExecuteMsg::ModifyChainChannelLinks { operations } => {
+            execute::connection_operations(deps, operations)
+        }
+    }
+}
+
+#[cfg_attr(not(feature = "imported"), entry_point)]
+pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
+    deps.api.debug(&format!("executing  query: {msg:?}"));
+    match msg {
+        QueryMsg::GetAddressFromAlias { contract_alias } => {
+            to_binary(&CONTRACT_ALIAS_MAP.load(deps.storage, &contract_alias)?)
+        }
+
+        QueryMsg::GetDestinationChainFromSourceChainViaChannel {
```

Comment:
very clear name! :)

Reply:
Slightly verbose but Ill take it for the clarity 😅